### ISSUE TITLE
Filter lose focus while typing

### ISSUE THREAD

**Issue by Silimim** [ghiscoding/Angular-Slickgrid#1475]
### Describe the bug

We've found out a problem with filters input and modals, after a modal is opened and closed filters lose their focus after keydown.

**Observations:**
It seems to happens when dataview changes after filtering, if rows do not change, input don't lose focus.

**What i've tried:**
- Invalidate the grid
- Reinit filter service
- Set display none and after time set back it's display property
- ResizeGrid method from resize service
- Refresh on DataviewObj

![GESI-Cortis-Lentini-S r l-Google-Chrome-2024-09-19-12-22-19](https://github.com/user-attachments/assets/b9010839-2978-42fb-8292-04f0a4aea5a6)

**Most close solution:**
The only workarout i've found is setting `filterTypingDebounce: 750` this allows to write more chars, but after debounce end input still lose focus

### Reproduction

**Steps:**
- Open a BSModal from a button in grid
- Close it
- Type something on filters

### Environment Info

```shell
**Versions:**
angular-slickgrid: 7.7.0
angular: 17.3.11
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/ghiscoding/Angular-Slickgrid/blob/master/.github/CODE_OF_CONDUCT.md)
- [X] Read the [HOWTO - Quick Start](https://ghiscoding.gitbook.io/angular-slickgrid/getting-started/quick-start).
- [X] Check that there isn't [already an issue](https://github.com/ghiscoding/Angular-Slickgrid/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/ghiscoding/Angular-Slickgrid/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ghiscoding** [comment]
Well I see that as a normal behavior since you clicked on the button in grid, you changed the focus to the cell

**Comment by Silimim** [comment]
Sorry maybe i didn't describe it well. I meant the problem is after closing the modal. Clicking on filter input. After the first char the filter loses focus.

The gif was a bit too long and dispersive, sorry.

**Comment by ghiscoding** [comment]
I can't help without a full repro

**Comment by Silimim** [comment]
Sorry for the late response, i'm hardly trying to repro the error but is harder than i thought. It seems some sort of strange interaction between slickgrid and ngx-bootstrap modals.
While trying to repro i've found out that after closing the modal and typing something an element is getting focused.

![image](https://github.com/user-attachments/assets/e3868509-bc3c-4d84-a2c1-2f276a6eaf06)

Which is one of the following two:

![image](https://github.com/user-attachments/assets/92bf4643-c8af-4329-957c-d69e60550518)

Deleting those elements filters focus works as expected after dataview changes.
Now i'm looking for a better way to prevent this. Also trying to make the repro.

**Comment by ghiscoding** [comment]
This seems to be more of an upstream dependency problem unrelated to my project and with this new info, if you search on the ngx-bootstrap project you will find similar issues like this one

[Longer closing time of modals in 18.0.2 affecting Cypress tests](https://github.com/valor-software/ngx-bootstrap/issues/6678)
[Modal (feat): On close should return focus to launching element](https://github.com/valor-software/ngx-bootstrap/issues/6259)

and if you look at some of their implementation on this [line](https://github.com/valor-software/ngx-bootstrap/blob/d210ca6dcdf213bfb63f4499ceffd258b3cd5a68/src/modal/modal-container.component.ts#L162-L177) of code, you see that they have a `setTimeout` that runs after the closing with a delay set by the modal transition which is maybe why the focus is lost after a quick time.

So it really looks like this is unrelated to my project and ngx-boostrap is playing with the set focus and taking a look at the default filter input in Slickgrid-Universal, it doesn't have any set focus. So again, it all seems to point at ngx-bootstrap being the culprit and I certainly don't support, neither want to troubleshoot external libraries. As per the first issue above, maybe give a try to fix your version of ngx-bootstrap at v18.0.0 (not v18.0.2 because of the issue) and/or maybe try to disable the animation and see if that helps... in summary, I more inclined to close this issue since it all seems to point at the upstream (external) dependency

**Comment by Silimim** [comment]
I'm pretty sure too that one of the main actors of this problem is ngx-bootstrap but there's some things i'm struggling with.

Filters get blurred not after time (waiting 1 to 10 seconds before typing is not different). But exactly when search result is diffrent from initial result.

For example:

In 10 rows, all containing the char 'R' filter is not losing focus.
In 10 rows, only 6/10 containing the char 'R' filter is not losing focus.

Also the DOM activeElement is inside anglar-slickgrid DOM element (as mentioned above). And deleting and reinserting it resolve the problem.

**Comment by ghiscoding** [comment]
Well the thing is that you still haven't provided any repro and if you can't provide a repro then you are on your own to debug it. If I take a look at the Slickgrid-Universal `inputFilter.ts` file, I do have a `blur` event listener but this listener is shared by these 3 events (blur, keyup, change) and so adding a breakpoint into that event is a bit tricky. You can add a conditional breakpoint as shown in animated gif below and then look at the call stack to find out who or what triggered the blur and if you're lucky it might help. The other bad thing is that Angular Zone.js wraps all events into a ton of wrappers and it's hard to find the actual listener which is why I went directly to the `inputFilter` file but that might not the actual culprit.

I seriously can't help without a full repro which you did not provide, so unless there's a repro, I will eventually close this issue. This project is provided for (Open Source) but I certainly don't want to lose a ton of time troubleshooting something without a repro. 

![Code_-_Insiders_gmxBUeyeXR](https://github.com/user-attachments/assets/febdea6f-d424-4174-b508-ef961de41160)

You can also see this SO to guide on how to add a conditional breakpoint
[JavaScript :How to set a Conditional Break Point in Chrome debugger tools](https://stackoverflow.com/questions/14598561/javascript-how-to-set-a-conditional-break-point-in-chrome-debugger-tools)

**Comment by ghiscoding** [comment]
@Silimim I'm pretty sure that I've fixed this problem in latest release via this Slickgrid-Universal PR [fix: input filter loses focus after edit+filtering with `enableExcelCopyBuffer`](https://github.com/ghiscoding/slickgrid-universal/pull/1750). I found that the issue was occurring when `enableExcelCopyBuffer` was enabled and an edit cell was clicked then typing in a filter would lose its focus after the first character typing. I found this bug in the Example 3 live demo while testing something else.

So the latest release v8.10.2 includes the fix for this issue (assuming this is the same issue which is likely)
